addons/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/handler/RFXComBridgeHandler.java

```diff
@@ -376,6 +288,40 @@ public void packetReceived(byte[] packet) {
                     if (msg.subType == SubType.RESPONSE) {
                         logger.debug("RFXCOM transceiver/receiver type: {}, hw version: {}.{}, fw version: {}",
                                 msg.transceiverType, msg.hardwareVersion1, msg.hardwareVersion2, msg.firmwareVersion);
+
+                        if (msg.command == Commands.GET_STATUS) {
+                            if (configuration.ignoreConfig) {
+                                logger.debug("Ignoring transceiver configuration");
+                            } else {
+                                byte[] setMode = new byte[0];
+
+                                try {
+                                    if (configuration.setMode != null && !configuration.setMode.isEmpty()) {
+                                        setMode = DatatypeConverter.parseHexBinary(configuration.setMode);
+                                    } else {
+                                        setMode = createConfMessage(msg.transceiverType);
+                                    }
+                                } catch (IllegalArgumentException ee) {
+                                    logger.warn("Failed to parse setMode data", ee);
```

Comment:
imho if this error occurs why will never enter the if block, should we embed the if in there as well? Then we don't need to assign the initial value to setMode